waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -501,6 +504,8 @@ method init*(g: OnchainGroupManager): Future[void] {.async.} =
     except CatchableError:
       error "failed to restart group sync", error = getCurrentExceptionMsg()
 
+  # Contract starts from the first index
```

Comment:
Why are we calculating the startIndex manually (setting g.latestIndex to be `1` here, and then using that value in `atomicBatch`, instead of passing the index retrieved from the event logs?

I'm probably wrong, so please correct me if that's the case, but I think this might be problematic in case of a partial sync or if a lot of time has passed since the last sync,  since in `getAndHandleEvents` the function `setMetadata` is called for each block chunk, and if for some reason we don't sync the full chain, the next time we start nwaku,  it will load the metadata to retrieve the latest block that was synced, and will call atomicOperation passing the index `1`, instead of the index from the logs that were emitted in the blocks that are being processed, thus the leaves from the merkle tree will be overwritten instead of inserting new elements.

Reply:
Hi, thanks for this issue - fixed in 048cdc29